File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/RegexUtils.cpp
#include <string>
#include <vector>
#include <regex>

class RegexUtils {
public:
    bool match(const std::string& pattern, const std::string& text) {
        std::regex e(pattern);
        std::smatch m;
        if (std::regex_search(text, m, e)) {
            return m.position() == 0;
        }
        return false;
    }

    std::vector<std::string> findall(const std::string& pattern, const std::string& text) {
        std::regex e(pattern);
        std::vector<std::string> result;
        auto it = std::sregex_iterator(text.begin(), text.end(), e);
        auto end = std::sregex_iterator();
        for (; it != end; ++it) {
            result.push_back(it->str());
        }
        return result;
    }

    std::vector<std::string> split(const std::string& pattern, const std::string& text) {
        std::regex e(pattern);
        std::vector<std::string> result;
        std::regex_token_iterator<std::string::const_iterator> it(text.begin(), text.end(), e, -1);
        std::regex_token_iterator<std::string::const_iterator> end;
        for (; it != end; ++it) {
            result.push_back(it->str());
        }
        return result;
    }

    std::string sub(const std::string& pattern, const std::string& replacement, const std::string& text) {
        std::regex e(pattern);
        return std::regex_replace(text, e, replacement);
    }

    std::string generate_email_pattern() {
        return R"(\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b)";
    }

    std::string generate_phone_number_pattern() {
        return R"(\b\d{3}-\d{3}-\d{4}\b)";
    }

    std::string generate_split_sentences_pattern() {
        return R"([.!?][\s]{1,2}(?=[A-Z]))";
    }

    std::vector<std::string> split_sentences(const std::string& text) {
        std::string pattern = generate_split_sentences_pattern();
        return split(pattern, text);
    }

    bool validate_phone_number(const std::string& phone_number) {
        std::string pattern = generate_phone_number_pattern();
        return match(pattern, phone_number);
    }

    std::vector<std::string> extract_email(const std::string& text) {
        std::string pattern = generate_email_pattern();
        return findall(pattern, text);
    }
};

File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/cpp_to_py/RegexUtils.py
import re


class RegexUtils:
    @staticmethod
    def match(pattern, text):
        return re.search(pattern, text) is not None

    @staticmethod
    def findall(pattern, text):
        return re.findall(pattern, text)

    @staticmethod
    def split(pattern, text):
        if text == "":
            return []
        tokens = re.split(pattern, text)
        if tokens[0] != text:
            tokens.insert(0, "")
        return tokens

    @staticmethod
    def sub(pattern, replacement, text):
        return re.sub(pattern, replacement, text)

    @staticmethod
    def generate_email_pattern():
        return r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b'

    @staticmethod
    def generate_phone_number_pattern():
        return r'\b\d{3}-\d{3}-\d{4}\b'

    @staticmethod
    def generate_split_sentences_pattern():
        return r'(?<=[.!?])\s+(?=[A-Z])'

    @staticmethod
    def split_sentences(text):
        pattern = RegexUtils.generate_split_sentences_pattern()
        sentences = RegexUtils.split(pattern, text)
        if sentences and sentences[0] == '':
            sentences.pop(0)
        if sentences and sentences[-1] == '':
            sentences.pop()
        return sentences

    @staticmethod
    def validate_phone_number(phone_number):
        pattern = RegexUtils.generate_phone_number_pattern()
        return RegexUtils.match(pattern, phone_number)

    @staticmethod
    def extract_email(text):
        pattern = RegexUtils.generate_email_pattern()
        return RegexUtils.findall(pattern, text)